src/plugins/code/queries.js

```diff
@@ -0,0 +1,10 @@
+// Copyright 2018 OpenStax Poland
+// Licensed under the MIT license. See LICENSE file in the project root for
+// full license text.
+
+export function getActiveCode(editor, value) {
+  const block = value.startBlock
+  if (!block) return null
+
+  return  block.type === 'code' ? block : null
```

Comment:
I think `getActiveSth` queries should only be created for blocks where this operations is not trivial. Admonitions, exercises, lists, and sections have them because selection may be arbitrarily deep inside them, and so finding whether you are inside one of them is not trivial (which just made me realise that `getActiveExercise` and `getActiveAdmonition` are broken as they assume at most one level of nesting, but can contain lists...). Here isn't any such reason.